File: backend/src/utils/logger.js
// Minimal, production-focused logger for backend/API/database
// DO NOT log sensitive data or debug info

const logger = {
  request: (method, url) => {
    console.log(`[REQUEST] ${method} ${url}`);
  },
  response: (status, method, url, ms) => {
    console.log(`[RESPONSE] ${status} ${method} ${url} (${ms}ms)`);
  },
  error: (message, method, url) => {
    console.error(`[ERROR] ${message} (${method} ${url})`);
  },
  dbQuery: (table, action) => {
    console.log(`[DB QUERY] ${table}/${action}`);
  },
  dbResponse: (table, action, success, ms) => {
    console.log(`[DB RESPONSE] ${success ? 'success' : 'failure'} ${table}/${action} (${ms}ms)`);
  },
  dbError: (message, table, action) => {
    console.error(`[DB ERROR] ${message} (${table}/${action})`);
  }
};

module.exports = logger;

File: mobile/lib/core/logger.dart
/// Minimal, production-focused logger for API calls only.
/// DO NOT log UI events, sensitive data, or debug info.
class Logger {
  Logger._();
  static final Logger instance = Logger._();

  void apiRequest(String endpoint) {
    // ignore: avoid_print
    print('[API REQUEST] $endpoint');
  }

  void apiResponse(int status, String endpoint) {
    // ignore: avoid_print
    print('[API RESPONSE] $status $endpoint');
  }

  void apiError(String message, String endpoint) {
    // ignore: avoid_print
    print('[API ERROR] $message ($endpoint)');
  }

  void apiWarning(String message, String endpoint) {
    // ignore: avoid_print
    print('[API WARNING] $message ($endpoint)');
  }
}
